File: fastframe/src/main/java/com/code/fastframe/baseactivity/BaseTitleActivity.java
package com.code.fastframe.baseactivity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.Nullable;
import com.code.fastframe.R;
import com.code.fastframe.ciface.IBaseLayout;
import com.code.fastframe.ciface.ITitle;
import com.gyf.immersionbar.ImmersionBar;

/**
 * Created by dengshaomin on 2017/11/7.
 */

public abstract class BaseTitleActivity extends BaseBundleActivity implements IBaseLayout, ITitle {

  protected ViewGroup container;

  private View left_image, right_image, nav_status_space;

  private TextView title_text;

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_title_base);
    initTitle();
    this.initView();
    this.initData();
    this.getNetData();
  }

  public void getNetData() {
  }

  @Override
  public void initData() {

  }

  private void initTitle() {
    container = findViewById(R.id.container);
    if (setContentLayout() != 0) {
      LayoutInflater.from(this).inflate(setContentLayout(), container);
    }
    title_text = findViewById(R.id.title_text);
    left_image = findViewById(R.id.left_image);
    right_image = findViewById(R.id.right_image);
    nav_status_space = findViewById(R.id.nav_status_space);
    left_image.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        titleLeftClick();
      }
    });
    right_image.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        titleRightClick();
      }
    });
    title_text.setText(TextUtils.isEmpty(setTitleText()) ? "" : setTitleText());
    if (setTitleLeftImage() > 0) {
      left_image.setBackgroundResource(setTitleLeftImage());
    }
    if (setTitleRightImage() > 0) {
      left_image.setBackgroundResource(setTitleRightImage());
    }
    if (needImmerse()) {
      ViewGroup.LayoutParams layoutParams = nav_status_space.getLayoutParams();
      layoutParams.height = ImmersionBar.getStatusBarHeight(this);
      nav_status_space.setLayoutParams(layoutParams);
      ImmersionBar.with(this).init();
    }
  }

  @Override
  public void titleLeftClick() {
    onBackPressed();
  }

  @Override
  public int setTitleLeftImage() {
    return 0;
  }

  @Override
  public int setTitleRightImage() {
    return 0;
  }

  @Override
  public void titleRightClick() {

  }

  @Override public boolean needImmerse() {
    return false;
  }
}

File: fastframe/src/main/java/com/code/fastframe/baseactivity/BaseTitleActivity.kt
package com.code.fastframe.baseactivity

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import com.code.fastframe.R.layout
import com.code.fastframe.ciface.IBaseLayout
import com.code.fastframe.ciface.ITitle
import com.gyf.immersionbar.ImmersionBar
import kotlinx.android.synthetic.main.activity_title_base.container
import kotlinx.android.synthetic.main.activity_title_base.left_image
import kotlinx.android.synthetic.main.activity_title_base.nav_status_space
import kotlinx.android.synthetic.main.activity_title_base.right_image
import kotlinx.android.synthetic.main.activity_title_base.title_text

/**
 * Created by dengshaomin on 2017/11/7.
 */
abstract class BaseTitleActivity : BaseBundleActivity(), IBaseLayout, ITitle {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(layout.activity_title_base)
    initTitle()
    initView()
    initData()
    requestNetData()
  }

  open fun requestNetData(){

  }

  override fun initData() {}
  private fun initTitle() {
    if (setContentLayout() != 0) {
      LayoutInflater.from(this).inflate(setContentLayout(), container)
    }
    left_image.setOnClickListener(View.OnClickListener { titleLeftClick() })
    right_image.setOnClickListener(View.OnClickListener { titleRightClick() })
    title_text.setText(if (TextUtils.isEmpty(setTitleText())) "" else setTitleText())
    if (setTitleLeftImage() > 0) {
      left_image.setBackgroundResource(setTitleLeftImage())
    }
    if (setTitleRightImage() > 0) {
      left_image.setBackgroundResource(setTitleRightImage())
    }
    if (needImmerse()) {
      val layoutParams = nav_status_space.getLayoutParams()
      layoutParams.height = ImmersionBar.getStatusBarHeight(this)
      nav_status_space.setLayoutParams(layoutParams)
      ImmersionBar.with(this).init()
    }
  }

  override fun titleLeftClick() {
    onBackPressed()
  }

  override fun setTitleLeftImage(): Int {
    return 0
  }

  override fun setTitleRightImage(): Int {
    return 0
  }

  override fun titleRightClick() {}
  override fun needImmerse(): Boolean {
    return false
  }

  override fun onDestroy() {
    super.onDestroy()
  }
}